src/fetchtastic/download/async_downloader.py

```
@@ -0,0 +1,492 @@
+        Returns:
+            bool: True if download succeeded, False otherwise.
+        """
+        max_retries = max_retries or self._get_max_retries()
+        delay = retry_delay or self._get_retry_delay()
+
+        for attempt in range(max_retries + 1):
+            try:
+                result = await self.async_download(url, target_path, progress_callback)
+                if result:
+                    return True
+            except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)